plutus-tx-plugin/test/Budget/monadicDo.plc.golden

```diff
@@ -66,7 +66,15 @@
           (lam
             y
             (con integer)
-            [ { Just (con integer) } [ [ (builtin addInteger) x ] y ] ]
+            (let
```

Comment:
As far as I can tell, the root cause of this additional let-binding is the same as `patternMatchDo.plc.golden`, where the original CoreExpr contains a `$fApplicativeMaybe_$cpure` instead of a `Just`. This leads to an additional `TermBind` in the initial PIR:
```
(termbind
  (strict)
  (vardecl fApplicativeMaybe_cpure (all a (type) (fun a [ Maybe a ])))
  (abs a (type) (lam ds a [ { Just a } ds ]))
)
```
 
This test case runs the PIR simplifier, and in the end I think this is what leads to this additional `ds` binding. `ds` isn't inlined because it is strict, and the RHS isn't pure and `ds` isn't the first subterm evaluated in the RHS.

Reply:
Worth trying to engineer a specific test case demonstrating the problem here? With a note containing this explanation?

Reply:
This test case is pretty small, I don't think it's worth the effort of manually shrinking it further. I added a comment next to the test case.